js/upgrade.js

```diff
@@ -801,7 +804,7 @@ $(document).ready(function() {
         var archive_prestashop = $("select[name=archive_prestashop]").val();
         var archive_num = $("input[name=archive_num]").val();
         var archive_xml = $("select[name=archive_xml]").val();
-        if (archive_num == "") {
+        if (archive_num == "" || (archive_num !== undefined && !archive_num.match(versionNumberRegex))) {
```

Comment:
actually no, but I understand your confusion:

this is for the case where the archive num field is not present, in this case we don't want to try the match, because the value is undefined (before adding this check, js would break because we tried a match on an undefined)